app/cluster/dynamic.go

```diff
@@ -0,0 +1,104 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rudderlabs/rudder-server/utils/types/servermode"
+)
+
+type modeProvider interface {
+	ServerMode() <-chan servermode.ModeAck
+}
+
+type Lifecycle interface {
+	Start()
+	Stop()
+}
+
+type Dynamic struct {
+	Provider modeProvider
```

Comment:
This is interesting... a public field of private type! Wonder how this would behave in practice :)

Reply:
We can export it ok, but no one should actually use this interface. 
Since will create coupling with this component dependency.